jbpm-runtime-manager/src/main/java/org/jbpm/runtime/manager/impl/PerCaseRuntimeManager.java

```diff
@@ -116,66 +115,31 @@ public RuntimeEngine getRuntimeEngine(Context<?> context) {
         if (!(context instanceof ProcessInstanceIdContext || context instanceof CaseContext)) {
             logger.warn("ProcessInstanceIdContext or CaseContext shall be used when interacting with PerCase runtime manager");
         }
-        
-        if (engineInitEager) {
-            KieSession ksession = null;
-            Long ksessionId = null;
 
-            RuntimeEngine localRuntime = findLocalRuntime(contextId);
-            if (localRuntime != null) {
-                return localRuntime;
-            }
-            synchronized (this) {
-                ksessionId = mapper.findMapping(context, this.identifier);
-                if (ksessionId == null) {
-                    ksession = factory.newKieSession();
-                    ksessionId = ksession.getIdentifier();
-
-                    if (context instanceof CaseContext) {
-                        ksession.execute(new SaveMappingCommand(mapper, context, ksessionId, getIdentifier()));
-                    }
-                } else {
-                    ksession = factory.findKieSessionById(ksessionId);
-                }
-            }
+        RuntimeEngine localRuntime = findLocalRuntime(contextId);
+        if (localRuntime != null) {
+            logger.debug("Runtime engine found in local {} with context {}", localRuntime, contextId);
+            return localRuntime;
+        }
+
+        Long ksessionId = mapper.findMapping(context, this.getIdentifier());
+        if(ksessionId == null && mapper instanceof InternalMapper) {
+            // we fall back as process instance must be closed but not really case and kie session must be still there.
+            logger.debug("PerCaseRuntimeManager KieSessionId found fallingback for context {} and owner {}", context.getContextId(), getIdentifier());
```

Comment:
Trace message is not clear, it should state that kieSessionId is null (or not found)

Reply:
changed to not found.